src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -400,7 +410,100 @@ private IObjectCreationExpression CreateBoundObjectCreationExpressionOperation(B
+        }
+
+        private IObjectOrCollectionInitializerExpression CreateBoundObjectInitializerExpressionOperation(BoundObjectInitializerExpression boundObjectInitializerExpression)
+        {
+            Lazy<ImmutableArray<IOperation>> initializers = new Lazy<ImmutableArray<IOperation>>(() => BoundObjectCreationExpression.GetChildInitializers(boundObjectInitializerExpression).SelectAsArray(n => Create(n)));
+            bool isInvalid = boundObjectInitializerExpression.HasErrors;
+            SyntaxNode syntax = boundObjectInitializerExpression.Syntax;
+            ITypeSymbol type = boundObjectInitializerExpression.Type;
+            Optional<object> constantValue = ConvertToOptional(boundObjectInitializerExpression.ConstantValue);
+            return new LazyObjectOrCollectionInitializerExpression(initializers, isInvalid, syntax, type, constantValue);
+        }
+
+        private IObjectOrCollectionInitializerExpression CreateBoundCollectionInitializerExpressionOperation(BoundCollectionInitializerExpression boundCollectionInitializerExpression)
+        {
+            Lazy<ImmutableArray<IOperation>> initializers = new Lazy<ImmutableArray<IOperation>>(() => BoundObjectCreationExpression.GetChildInitializers(boundCollectionInitializerExpression).SelectAsArray(n => Create(n)));
+            bool isInvalid = boundCollectionInitializerExpression.HasErrors;
+            SyntaxNode syntax = boundCollectionInitializerExpression.Syntax;
+            ITypeSymbol type = boundCollectionInitializerExpression.Type;
+            Optional<object> constantValue = ConvertToOptional(boundCollectionInitializerExpression.ConstantValue);
+            return new LazyObjectOrCollectionInitializerExpression(initializers, isInvalid, syntax, type, constantValue);
+        }
+
+        private IMemberReferenceExpression CreateBoundObjectInitializerMemberOperation(BoundObjectInitializerMember boundObjectInitializerMember)
+        {
+            Lazy<IOperation> instance = new Lazy<IOperation>(() => new InstanceReferenceExpression(InstanceReferenceKind.Implicit, boundObjectInitializerMember.HasErrors, syntax: boundObjectInitializerMember.Syntax, type: boundObjectInitializerMember.MemberSymbol.ContainingType, constantValue: default(Optional<object>)));
+            bool isInvalid = boundObjectInitializerMember.HasErrors;
+            SyntaxNode syntax = boundObjectInitializerMember.Syntax;
+            ITypeSymbol type = boundObjectInitializerMember.Type;
+            Optional<object> constantValue = ConvertToOptional(boundObjectInitializerMember.ConstantValue);
+
+            switch (boundObjectInitializerMember.MemberSymbol.Kind)
+            {
+                case SymbolKind.Field:
+                    var field = (FieldSymbol)boundObjectInitializerMember.MemberSymbol;
+                    return new LazyFieldReferenceExpression(field, instance, field, isInvalid, syntax, type, constantValue);
```

Comment:
Is the second `field` parameter necessary for `LazyFieldReferenceExpression`? (Are the two parameter values ever different?) Same question for `LazyEventReferenceExpression` and `LazyPropertyReferenceExpression`.